lib/yt/models/resource.rb

```diff
@@ -12,7 +12,13 @@ class Resource < Base
 
       # @!attribute [r] id
       #   @return [String] the ID that YouTube uses to identify each resource.
-      attr_reader :id
+      def id
+        if @url
```

Comment:
Do we need this if/else here? In your code if you don't instantiate with a url, you've got `@id` set already.